src/main/java/io/opencensus/integration/jdbc/Observability.java

```diff
@@ -228,6 +230,27 @@ private void recordStatWithTags(double value, TagContext tagContext) {
     }
   }
 
+  // Makes sure the tag value will be accepted as a valid tag value by TagValue::create
+  // This will limit the string to TagValue.MAX_LENGTH and if any character is not in
+  // the
+  private static String makeValidTagValue(String tagValue) {
+    int length = Math.min(tagValue.length(), TagValue.MAX_LENGTH);
+    StringBuilder validTagValue = new StringBuilder(length);
+    for (int i = 0; i < length; i++) {
+      char ch = tagValue.charAt(i);
+      if (isPrintableChar(ch)) {
+        validTagValue.append(ch);
+      } else {
+        validTagValue.append(' ');
+      }
+    }
+    return validTagValue.toString();
+  }
+
+  private static boolean isPrintableChar(char ch) {
```

Comment:
This is copied from io.opencensus.internal.StringUtils.isPrintableString which alas does not expose this method

Reply:
Ack, I hear ye!